### PR TITLE
Feature 267 db engine

### PR DESCRIPTION
**Pull request by FlorianK13**
Created an engine parameter at Mastr __init__(). This parameter decides which database is used.

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -39,9 +39,9 @@ API download
 -----------------------------------
 When using `download(method="API")`, the data is retrieved from the MaStR API. For using the MaStR API,
 credentials are needed (see :ref:`Get data via the MaStR-API`). By using the API,
-additional parameters can be set to define in detail which data should be reveived.
+additional parameters can be set to define in detail which data should be reviewed.
```
Comment (chrwm) [inline docs/getting_started.rst]:
```suggestion
additional parameters can be set to define in detail which data should be obtained.
```

Comment (chrwm) [inline docs/getting_started.rst]:
```suggestion
additional parameters can be set to define in detail which data should be obtained.
```

File: open_mastr/mastr.py [hunk 2]
```diff
@@ -1,10 +1,8 @@
-from datetime import date, datetime
+from datetime import date
```
Comment (chrwm) [inline open_mastr/mastr.py:1]: Clear commit message that fits changes, nice and easy to review!

File: open_mastr/utils/helpers.py [hunk 3]
```diff
@@ -56,16 +56,23 @@ def create_database_engine(engine):
...
+def validate_parameter_format_for_mastr_init(engine) -> None:
+    if engine not in ["sqlite", "docker-postgres"]:
+        if not isinstance(engine, sqlalchemy.engine.Engine):
+            raise ValueError(
+                "parameter engine has to be either 'sqlite' "
+                "or 'docker-postgres' or an sqlalchemy.engine.Engine object."
+            )
+
+
 def validate_parameter_format_for_download_method(
```
Comment (chrwm) [inline open_mastr/utils/helpers.py]: looks fine!

File: tests/test_helpers.py [hunk 4]
```diff
@@ -178,3 +183,15 @@ def get_parameters_from_parameter_dict(parameter_dict):
...
+def test_validate_parameter_format_for_mastr_init(db):
+    engine_list_working = ["sqlite", "docker-postgres", db._engine]
+    engine_list_failing = ["HI", 12]
+
+    for engine in engine_list_working:
+        assert validate_parameter_format_for_mastr_init(engine) is None
+
+    for engine in engine_list_failing:
+        with pytest.raises(ValueError):
+            validate_parameter_format_for_mastr_init(engine)
```
Comment (chrwm) [inline tests/test_helpers.py:201]: Is this implementation resulting in a test failure for all other values apart from `["sqlite", "docker-postgres", db._engine]` and not only a failure for `["HI", 12]` ?

File: docs/advanced.rst [hunk 5]
```diff
@@ -51,8 +51,14 @@ Data
...
-The SQLite database is stored in `$HOME/.open-MaStR/data/sqlite`, unless a specific database engine is passed during
-:class:`open_mastr.mastr.Mastr` initialization.
+
+The data can then be written to an sql database. The type of the sql databse is determined 
+by the parameter `engine` in the Mastr class (see :ref:`mastr module`). The possible databases are:
+
+* slqite: This database will be stored in `$HOME/.open-MaStR/data/sqlite`.
+* docker-postgres: A docker container of a PostgreSQL database. 
+* own database: The Mastr class accepts a sqlalchemy.engine.Engine object as engine which provides the user to 
+  use any other desired database.
```
Comment (chrwm) [inline docs/advanced.rst]: Thanks for adding this!

File: tests/test_helpers.py (hunk 4 above)
Comment (FlorianK13) [reply tests/test_helpers.py:201]: Yes, but it is hard to test for such a large amount of failing values, so I just chose an exemplary string and integer value that makes the code fail.

**Comment by FlorianK13** [comment]
@chrwm Can we merge the branch now? Because Deniz and me want to do a restructuring session later today and it would be best if this branch is already merged then.

**Review by deniztepe — APPROVED** [review]
Checked

**Pull request merged**